crates/holochain_persistence_api/src/reporting.rs

```diff
@@ -0,0 +1,12 @@
+use super::error::{PersistenceError, PersistenceResult};
+
+pub trait ReportStorage {
+    /// Return the number of bytes this storage implementation is using on the host system.
+    /// The actual implementation is up to the author of the persistence implementation
+    /// and may be disk usage or memory usage
+    fn get_byte_count(&self) -> PersistenceResult<usize> {
+        Err(PersistenceError::ErrorGeneric(
+            "Not implemented for this storage type".into(),
+        ))
+    }
+}
```

Comment:
Is byte count enough info for a first cut? I would suggest maybe future proofing this by returning a struct called `StorageReport` which contains just one field equal to byte count. Open to other suggestions or a defense as to why the current version is better though.

Reply:
Yes, I agree with @struktured we may want other stats on the performance of the particular cas implementation being used, and other breakdowns on the data, so yah lets add a struct...